builder/js/ncf.js

```diff
@@ -653,13 +672,24 @@ app.controller('ncf-builder', function ($scope, $modal, $http, $log, $location,
 
   $scope.setBundleName = function (technique) {
     if (technique.bundle_name === undefined) {
-      // Replace all non alpha numeric character (\W is [^a-zA-Z_0-9]) by _
-      var bundle_name = technique.name.replace(/\W/g,"_");
-      technique.bundle_name = bundle_name;
+      technique.bundle_name =  $scope.getBundleName(technique.name);
     }
     return technique;
   }
 
+  $scope.selectedTechniqueBundleName = "";
+
+  $scope.getBundleName = function (techniqueName) {
+    // Replace all non alpha numeric character (\W is [^a-zA-Z_0-9]) by _
+    return techniqueName ? techniqueName.replace(/\W/g,"_") : "";
+  }
+
+  $scope.updateBundleName = function () {
+    if($scope.selectedTechnique.bundle_name===undefined){
```

Comment:
Here you should do like you do in the validator and look in original technique !!